src/main/kotlin/io/github/mojira/arisa/modules/CrashModule.kt

```diff
@@ -40,28 +43,64 @@ class CrashModule(
                 .toMutableList()
             textDocuments.add(TextDocument(body ?: "", created))
 
-            val infos = textDocuments
+            val crashes = textDocuments
+                .asSequence()
                 .filter(::isTextDocumentRecent)
-                .mapNotNull(::fetchInfo)
-            assertNotEmpty(infos).bind()
+                .map { processCrash(it) }
+                .filter { it.second.isRight() }
+                .map { extractCrash(it) }
+                .filter { it.second is Crash.Minecraft || it.second is Crash.Java }
+                .toList()
 
-            val mostRelevantInfo = infos.reduce(::getMoreRelevantInfo)
+            assertNotEmpty(crashes).bind()
 
-            if (mostRelevantInfo.modded) {
+            if (crashes.any { it.second is Crash.Minecraft && (it.second as Crash.Minecraft).modded }) {
                 addModdedComment().toFailedModuleEither().bind()
                 resolveAsInvalid().toFailedModuleEither().bind()
             } else {
-                val configs = crashDupeConfigs.filter(::isConfigValid)
-                val key = getDuplicateKey(mostRelevantInfo, configs)
-                assertNotNull(key).bind()
+                val sortedMap = crashes.toMap().toSortedMap(compareByDescending { it.created })
+                val key = findDuplicate(sortedMap, crashDupeConfigs)
```

Comment:
I'm also not too sure it should try to find a duplicate in all crashes.
The current module only sees the most recent crash that is not above a certain age. This might be fine if the max attachment age is less then 30 days though (as supposed to the original js module). I'd say it would be good if it just used the lastRun time.

It might also be good to ignore tickets with a confirmation status, Mojang Priority, ADO or assignee, just in case someone attaches an irrelevant crash report to an otherwise valid ticket.

Reply:
This looks for any match in less than 30 days. I was thinking on the case that a user posts multiple logs(like launcher log and multiple crashes).

Agree with checking confirmation status + Mojang priority (not sure how you would check ADO)

Reply:
ADO is probably another custom field that would need a config entry.

Reply:
It would still prefer the most recent crash of all

Reply:
Added the confirmed and mojang priority check

Reply:
I would exclude the crash module from MCPE anyways

Reply:
This only operates in MC, MCL and MCTEST, we shouldnt need ADO.

Reply:
Oh yes, good point